includes/Services/FlowService.php

```diff
@@ -0,0 +1,203 @@
+							$flow_data[$fix['new_key']] = $flow_data[$fix['old_key']]
+							: $flow_data[$fix['new_key']] = $default_flow_data[$fix['new_key']];
+						unset($flow_data[$fix['old_key']]);
+					}	
+				}
+			}
+
+			if( ! array_key_exists($key, $flow_data)) {
+				$updated_flow_data[$key] = $value;
+				continue;
+			}
+
+			// Accepts the value of Exception List keys as is from the database OR non array OR empty array values
+			if(isset($exception_list[$key]) || ! is_array($value) || (is_array($value) && sizeof($value) === 0)) {
+				$updated_flow_data[$key] = $flow_data[$key];
+				continue;
+			}
+
+			// To handle Indexed Arrays gracefully
+			if (count(array_filter(array_keys($value), 'is_string')) === 0) {
+				// To check if an Indexed Array is further Nested or Not
+				foreach($value as $index_key => $index_value) {
+					// For Indexed Arrays having Non Associative Array Values
+					(!is_array($index_value) && isset($flow_data[$key]))? 
+						$updated_flow_data[$key] = $flow_data[$key] 
+						: $updated_flow_data[$key] = $value;
+				}
+				continue;
+			} 
+
+			// To handle Associative Arrays gracefully
+			$updated_flow_data[$key] = self::update_flow_data_recursive($value, $flow_data[$key]);
+		}
+		return $updated_flow_data;
+	}
+
+	/*
+	 * function to update the Database recursively based on Values opted or entered by the User
+	 */	
+	private static function update_post_call_data_recursive(&$flow_data, $params) {
+		$exception_list = Flows::get_exception_list();
+
+		foreach ($flow_data as $key => $value)
+		{ 		
+			if (! array_key_exists($key, $params)) {
+				$flow_data[$key] = $value;	
+				continue;
+			}
+
+			// Accepts the value of Exception List keys as is from the database
+			if($exception_list && isset($exception_list[$key])) {
+				$flow_data[$key] = $params[$key];
+				continue;
+			}
+
+			// Error thrown if the datatype of the parameter does not match
+			if(strcmp(gettype($value), gettype($params[$key])) != 0) {
+				return new \WP_Error(
+					'wrong_param_type_provided',
+					"Wrong Parameter Type Provided : ". $key. ' => '. gettype($params[$key]) . '. Expected: ' . gettype($value),
+					array( 'status' => 404 )
+				);
+			}
+
+			// Accepts non-Array Values entered by the user
+			if(!is_array($value)) {
+				$flow_data[$key] = $params[$key];
+				continue;
+			}
+
+			// To handle Indexed Arrays gracefully
+			if (count(array_filter(array_keys($params[$key]), 'is_string')) === 0 ) {
+				// If the Database value is empty or an Indexed Array, to avoid Associative arrays to be overwritten (Eg: data)
+				((sizeof($value) === 0) || (count(array_filter(array_keys($value), 'is_string')) === 0))?
+					$flow_data[$key] = $params[$key]
+					: $flow_data[$key] = $value;
+				continue;
+			}
+
+			// To handle Associative Arrays gracefully
+			$flow_data[$key] = self::update_post_call_data_recursive($value, $params[$key]);
+			if( \is_wp_error($flow_data[$key]) ) {
+				return $flow_data[$key];
+			}
```

Comment:
Let's not assign error to `$flow_data` since this is a referenced variable on line 95, let's assign only when it has a valid value.